dbms/src/Common/tests/gtest_UTF8Helpers.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <TestUtils/TiFlashTestBasic.h>
+
+#include <Common/UTF8Helpers.cpp>
+
+namespace DB
+{
+namespace tests
+{
+/// Reference https://github.com/skeeto/branchless-utf8/blob/master/test/tests.c
+
+unsigned char * utf8_encode(unsigned char * s, long c)
```

Comment:
Can we reuse code like `void utf8Encode(char * buf, size_t & used_length, UInt32 unicode)`

Reply:
The utf8_encode is used here to provide illegal characters for tests only.